extensions/wikia/CuratedContent/CuratedContentHelper.class.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * CuratedContent Helper
+ */
+class CuratedContentHelper {
+	const STR_ARTICLE = 'article';
+	const STR_BLOG = 'blog';
+	const STR_FILE = 'file';
+	const STR_CATEGORY = 'category';
+	const STR_EMPTY_CATEGORY = 'emptyCategory';
+
+	public function processLogic( $sections ) {
+		$processedSections = [ ];
+		foreach ( $sections as $section ) {
+			$processedSections[ ] = $this->processLogicForSection( $section );
+		}
+		return $processedSections;
+	}
+
+	private function processLogicForSection( $section ) {
+		$section['image_id'] = (int)$section['image_id'];
+
+		foreach ( $section['items'] as &$item ) {
+			$this->fillItemInfo( $item );
+		}
+
+		return $section;
+	}
+
+	private function fillItemInfo( &$item ) {
+		$title = Title::newFromText( $item['title'] );
+		if ( !empty( $title ) ) {
+			$articleId = $title->getArticleId();
+			$namespaceId = $title->getNamespace();
+			$imageId = (int)$item['image_id'];
+
+			$item['type'] = self::getTypeFromNamespaceId( $namespaceId );
+
+			switch ( $item['type'] ) {
```

Comment:
What if there is no case for that switch? Maybe default case should be there and return some exception or sth else?